lib/datadog/statsd/message_buffer.rb

```diff
@@ -92,6 +123,33 @@ def ensure_sendable!(message_size)
       def bytesize_threshold
         @bytesize_threshold ||= (max_payload_size - PAYLOAD_SIZE_TOLERANCE * max_payload_size).to_i
       end
+
+      def create_flush_thread
+        flush_thread = Thread.new(&method(:flush_loop))
+        flush_thread.name = 'Statsd MessageBuffer' if flush_thread.respond_to?(:name=)
+        flush_thread
+      end
+
+      def flush_loop
+        last_flush_time = current_time
+        until @closed
+          @mon.synchronize do
+            @cv.wait(flush_interval - (current_time - last_flush_time))
+            last_flush_time = current_time
+            flush
+          end
+        end
+      end
+
+      if Process.const_defined?(:CLOCK_MONOTONIC)
+        def current_time
+          Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        end
+      else
+        def current_time
+          Time.now
+        end
+      end
```

Comment:
`Statsd.now` will do this for you :)

Reply:
I didn't know that! It seems a private instance method, so I have to make it a public singleton method to use it from here. I'm afraid that some users call it from their code, that is, outside this library if I make it public. What do you think about that?

Reply:
Oh, hm, that's a good point.  Let's leave it as-is, then.